xonsh/procs/specs.py

```diff
@@ -442,6 +444,7 @@ def run(self, *, pipeline_group=None):
         kwargs = {n: getattr(self, n) for n in self.kwnames}
         self.prep_env(kwargs)
         if callable(self.alias):
+            kwargs["env"]["ALIAS_NAME"] = self.alias_name
```

Comment:
I'm fine with setting env vars for this, as long as we use names that are obviously internal like `__XONSH_ALIAS_NAME` and `__XONSH_ALIAS_STACK`